src/Symfony/Component/Lock/Store/RedisStore.php

```diff
@@ -239,9 +262,27 @@ private function evaluate(string $script, string $resource, array $args): mixed
         if ($this->redis instanceof \RedisArray) {
             $client = $this->redis->_instance($this->redis->_target($resource));
             $client->clearLastError();
-            $result = $client->eval($script, array_merge([$resource], $args), 1);
-            if (null !== $err = $client->getLastError()) {
-                throw new LockStorageException($err);
+            $result = $client->evalSha($scriptSha, array_merge([$resource], $args), 1);
+            if (null !== ($err = $client->getLastError()) && self::NO_SCRIPT_ERROR_MESSAGE === $err) {
+                $client->clearLastError();
+
+                if ($this->redis instanceof \RedisCluster) {
```

Comment:
this will be false. We cannot have both a RedisArray and a RedisCluster at the same time.